### PR TITLE
Patch ocaml-variants.5.1.0~rc3+tsan with ocaml/ocaml#12541

### PR DESCRIPTION
**Pull request by OlivierNicole**
Trying to install the switch with an old gcc or clang will fail with unhelpful error messages. This patches the switch with the fix from ocaml/ocaml#12541, which gives a useful error.

### CODE REVIEW

File: packages/ocaml-variants/ocaml-variants.5.1.0~rc3+tsan/opam [hunk 1]
```diff
@@ -47,7 +47,10 @@ url {
   src: "https://github.com/ocaml-multicore/ocaml-tsan/archive/5.1.0-rc3+tsan.tar.gz"
   checksum: "sha256=d5b80d65ace8912f060b14d552b06ea1df9ef28f0cd0e70a951bb5398fb0d491"
 }
-extra-files: ["ocaml-variants.install" "md5=3e969b841df1f51ca448e6e6295cb451"]
+extra-files: [
+  ["ocaml-variants.install" "md5=3e969b841df1f51ca448e6e6295cb451"]
+  ["check-tsan-distinguish-volatile.patch" "6e47bf51a2360805f49579884d51c29c"]
```
Comment (kit-ty-kate) [inline packages/ocaml-variants/ocaml-variants.5.1.0~rc3+tsan/opam:52]:
```suggestion
  ["check-tsan-distinguish-volatile.patch" "md5=6e47bf51a2360805f49579884d51c29c"]
```

**Comment by avsm** [comment]
Would it be possible to have these tsan variants hosted on a branch somewhere, rather than carrying the patches in opam-repo? Bear in mind that every user will have to download the patches if they are here, _vs_ only the people installing it if it's published in a branch.

**Comment by OlivierNicole** [comment]
I tried to do that, but it doesn’t seem to download the extra file. I took inspiration from other packages; not sure what I did wrong.

File: packages/ocaml-variants/ocaml-variants.5.1.0~rc3+tsan/opam [hunk 2]
```diff
@@ -47,7 +47,10 @@ url {
   src: "https://github.com/ocaml-multicore/ocaml-tsan/archive/5.1.0-rc3+tsan.tar.gz"
   checksum: "sha256=d5b80d65ace8912f060b14d552b06ea1df9ef28f0cd0e70a951bb5398fb0d491"
 }
-extra-files: ["ocaml-variants.install" "md5=3e969b841df1f51ca448e6e6295cb451"]
+extra-files: [
+  ["ocaml-variants.install" "md5=3e969b841df1f51ca448e6e6295cb451"]
+  ["check-tsan-distinguish-volatile.patch" "md5=6e47bf51a2360805f49579884d51c29c"]
+]
```
Comment (kit-ty-kate) [inline packages/ocaml-variants/ocaml-variants.5.1.0~rc3+tsan/opam:53]:
```suggestion
extra-files: ["ocaml-variants.install" "md5=3e969b841df1f51ca448e6e6295cb451"]
```

**Comment by avsm** [comment]
Thanks! You may want to announce this on <https://discuss.ocaml.org>, where we have a Community category and an `announce` tag for this purpose.

**Pull request merged**